src/main/java/no/sysco/middleware/metrics/prometheus/jdbc/config/Config.java

```diff
@@ -0,0 +1,68 @@
+package no.sysco.middleware.metrics.prometheus.jdbc.config;
+
+import static java.util.stream.Collectors.joining;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.exc.InvalidDefinitionException;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+
+/** Configuration for the Prometheus JDBC Exporter. */
+@ConfigObject
+@Value.Immutable
+@JsonDeserialize(builder = ImmutableConfig.Builder.class)
+public interface Config {
+
+    static Config parseYaml(InputStream data) throws IOException {
+        final var mapper = new ObjectMapper(new YAMLFactory());
+        mapper.registerModule(new Jdk8Module());
+        try {
+            return mapper.readValue(data, Config.class);
+        } catch (InvalidDefinitionException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        } catch (JsonMappingException e) {
+            throw new IllegalArgumentException(e.getMessage(), e);
+        } catch (JsonProcessingException e) {
```

Comment:
[InvalidDefinitionExcepiton][ide-jdoc] is more like a "bad annotation" thing, so it would rather indicate a bug in JDBC Exporter than bad config input...

[ide-jdoc]: https://github.com/FasterXML/jackson-databind/blob/jackson-databind-2.13.0/src/main/java/com/fasterxml/jackson/databind/exc/InvalidDefinitionException.java#L11-L15